modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/connectors/data/sql/SQLConnector.java

```diff
@@ -83,141 +83,150 @@ public Connection getSQLConnection() {
+            value = options.get(key);
+            if (value != null) {
+                config.setIsolateInternalQueries(Boolean.parseBoolean(value.stringValue()));
+                options.remove(key);
+            }
+            key = new BString(Constants.PoolProperties.ALLOW_POOL_SUSPENSION);
+            value = options.get(key);
+            if (value != null) {
+                config.setAllowPoolSuspension(Boolean.parseBoolean(value.stringValue()));
+                options.remove(key);
+            }
+            key = new BString(Constants.PoolProperties.READ_ONLY);
+            value = options.get(key);
+            if (value != null) {
+                config.setReadOnly(Boolean.parseBoolean(value.stringValue()));
+                options.remove(key);
+            }
+            key = new BString(Constants.PoolProperties.REGISTER_MBEANS);
+            value = options.get(key);
+            if (value != null) {
+                config.setRegisterMbeans(Boolean.parseBoolean(value.stringValue()));
+                options.remove(key);
+            }
+            key = new BString(Constants.PoolProperties.CATALOG);
+            value = options.get(key);
+            if (value != null) {
+                config.setCatalog(value.stringValue());
+                options.remove(key);
+            }
+            key = new BString(Constants.PoolProperties.CONNECTION_INIT_SQL);
+            value = options.get(key);
+            if (value != null) {
+                config.setConnectionInitSql(value.stringValue());
+                options.remove(key);
+            }
+            key = new BString(Constants.PoolProperties.DRIVER_CLASSNAME);
+            value = options.get(key);
+            if (value != null) {
+                config.setDriverClassName(value.stringValue());
+                options.remove(key);
+            }
+            key = new BString(Constants.PoolProperties.TRANSACTION_ISOLATION);
+            value = options.get(key);
+            if (value != null) {
+                config.setTransactionIsolation(value.stringValue());
+                options.remove(key);
+            }
+            key = new BString(Constants.PoolProperties.VALIDATION_TIMEOUT);
+            value = options.get(key);
+            if (value != null) {
+                config.setValidationTimeout(Long.parseLong(value.stringValue()));
+                options.remove(key);
+            }
+            key = new BString(Constants.PoolProperties.LEAK_DETECTION_THRESHOLD);
+            value = options.get(key);
+            if (value != null) {
+                config.setLeakDetectionThreshold(Long.parseLong(value.stringValue()));
+                options.remove(key);
+            }
+            setDataSourceProperties(options, config);
+            hikariDataSource = new HikariDataSource(config);
+        } catch (Throwable t) {
+            String errorMessage = "error in sql connector configuration";
+            if (t.getCause() != null) {
```

Comment:
T.getCause is used to avoid detailed exceptions which have java errors in some situations.